test/e2e/provisioning/testdata/machinedeployment-aws-assume-role.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: "cluster.k8s.io/v1alpha1"
```

Comment:
You don't need to use a new template, just use https://github.com/kubermatic/machine-controller/blob/master/test/e2e/provisioning/all_e2e_test.go#L55 and add those two fields to it: 
```
            assumeRoleARN: << AWS_ASSUME_ROLE_ARN >>
             assumeRoleExternalID: << AWS_ASSUME_ROLE_EXTERNAL_ID >>
``` 
helper functions will take care of the customising